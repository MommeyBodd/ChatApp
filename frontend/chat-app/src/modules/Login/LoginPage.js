import React, { useState, useEffect } from "react";
import Button from "@material-ui/core/Button";
import axios from "axios";
import PropTypes from "prop-types";

const LoginPage = ({}) => {
  const [isLoggedIn, changeStatus] = useState(null);
  console.log(isLoggedIn);
  return <div></div>;
};

LoginPage.propTypes = {};

export default LoginPage;
