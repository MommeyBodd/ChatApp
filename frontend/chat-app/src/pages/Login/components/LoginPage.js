import React, { useState, useEffect } from "react";
import Button from "@material-ui/core/Button";
import { useStyles } from "../styles/style";
import PropTypes from "prop-types";

const LoginPage = ({}) => {
  const classes = useStyles();
  return (
    <div className={classes.container}>
      <h1 className={classes.text}>Welcome to the Chat APP!</h1>
      <Button
        className={classes.button}
        color="secondary"
        variant="contained"
        href="http://localhost:3001/auth/google"
      >
        GOOGLE+
      </Button>
    </div>
  );
};

LoginPage.propTypes = {};

export default React.memo(LoginPage);
