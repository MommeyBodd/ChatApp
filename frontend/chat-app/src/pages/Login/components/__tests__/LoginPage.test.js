import React from "react";
import LoginPage from "../LoginPage";
import { createShallow } from "@material-ui/core/test-utils";
import Button from "@material-ui/core/Button";

describe("Login Page UI testing", () => {
  let shallow;
  let wrapper;

  beforeEach(() => {
    shallow = createShallow();
    wrapper = shallow(<LoginPage />);
  });

  test("should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("button should contain correct link", () => {
    const buttonLink = wrapper.find(Button).props().href;

    expect(buttonLink).toEqual("http://localhost:3001/auth/google");
  });
});
