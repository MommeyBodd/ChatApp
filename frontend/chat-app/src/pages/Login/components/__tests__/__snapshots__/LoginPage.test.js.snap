// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Page UI testing should render correctly 1`] = `
<div
  className="makeStyles-container-1"
>
  <h1
    className="makeStyles-text-2"
  >
    Welcome to the Chat APP!
  </h1>
  <WithStyles(ForwardRef(Button))
    className="makeStyles-button-3"
    color="secondary"
    href="http://localhost:3001/auth/google"
    variant="contained"
  >
    GOOGLE+
  </WithStyles(ForwardRef(Button))>
</div>
`;
